### PR TITLE
chore: refactor card meta to allow sharing it

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

lifted out of #12560 

Text and Insight Cards have a lot of overlap but could share more code

## Changes

* Lifts "meta" section out of the insight card so it can be used in text cards
* adds show details to text cards and hide modified by in their collapsed meta

|        |   |
|--------|---|
| before |  before-hid |
| after  |  after-hid |

|        |   |
|--------|---|
| before |  before-show|
| after  |  after-show|

## How did you test this code?

👀

### CODE REVIEW

File: frontend/src/lib/components/Cards/TextCard/TextCard.tsx [hunk 1]
```diff
@@ -61,90 +76,100 @@ export function TextCardInternal(
     const otherDashboards = nameSortedDashboards.filter((dashboard) => dashboard.id !== dashboardId)
     return (
         <div
-            className={clsx('TextCard rounded flex flex-col', className, showResizeHandles && 'border')}
+            className={clsx('Card TextCard rounded flex flex-col', className, showResizeHandles && 'border')}
```
Comment (Twixes) [inline frontend/src/lib/components/Cards/TextCard/TextCard.tsx:79]: Shouldn't we be using the `Card` component? Feels dirty to include a different component's class here.

File: frontend/src/lib/components/Cards/TextCard/TextCard.tsx [hunk 2]
```diff
@@ -61,90 +76,100 @@ export function TextCardInternal(
...
-                                    Edit text
-                                </LemonButton>
+            <CardMeta
+                showEditingControls={showEditingControls}
+                showDetailsControls={true}
+                setAreDetailsShown={setAreDetailsShown}
+                areDetailsShown={areDetailsShown}
+                className={clsx(showResizeHandles ? 'border-b' : 'border rounded-t')}
+                metaDetails={
+                    <UserActivityIndicator
```
Comment (Twixes) [inline frontend/src/lib/components/Cards/TextCard/TextCard.tsx:91]: The feedback about using the "Created by" format in the details part still holds
https://user-images.githubusercontent.com/4550621/199441563-6b5a1d59-526d-4a95-a3ba-2386eaf255fc.png

File: frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx [hunk 3]
```diff
@@ -217,266 +209,178 @@ function InsightMeta({
...
-                                <h5>
-                                    <span
-                                        title={
-                                            INSIGHT_TYPES_METADATA[filters.insight || InsightType.TRENDS]?.description
+                        </h4>
+                    </Link>
+
+                    <div className="CardMeta__description">{insight.description || <i>No description</i>}</div>
+                    {insight.tags && insight.tags.length > 0 && <ObjectTags tags={insight.tags} staticOnly />}
+                    <UserActivityIndicator at={insight.last_modified_at} by={insight.last_modified_by} />
```
Comment (Twixes) [inline frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx:244]: We should be consistent with this. If "Last modified by" is in the foldable section in text cards, it should be in the foldable section here too (next to "Created by")

**Review by Twixes — CHANGES_REQUESTED** [review]
The text's obscured a bit:
Screenshot 2022-11-07 at 13 32 30
I think the refactor makes sense, but there are some significant rough edges.

File: frontend/src/lib/components/Cards/Card.scss [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        max-width: 100%; // Make sure that horizontal overflow is capped, so that ellipsis on insight name works
+    }
+}
+
+.CardMeta__top {
+    align-self: stretch;
+    display: flex;
+    align-items: flex-start;
+    justify-content: space-between;
+    height: 2rem;
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply frontend/src/lib/components/Cards/Card.scss:122]: Screenshot 2022-11-08 at 22 19 12

Ah, I didn't spot this knock in... the text card needs a little bump to stop the CardMeta_controls from scrolling 🤔

File: frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx (hunk 3 above)
Comment (pauldambra) [reply frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx:244]: yep, will push this into #12560 wanted to keep this as small as I could get away with

File: frontend/src/lib/components/Cards/TextCard/TextCard.tsx (hunk 2 above)
Comment (pauldambra) [reply frontend/src/lib/components/Cards/TextCard/TextCard.tsx:91]: yep, will push this into #12560 wanted to keep this as small as I could get away with

File: frontend/src/lib/components/Cards/TextCard/TextCard.tsx (hunk 1 above)
Comment (pauldambra) [reply frontend/src/lib/components/Cards/TextCard/TextCard.tsx]: Is it better hidden in the scss? I can see us ending up with a factory that hides `InsightCard`, `TextCard`, and other kinds from the consumer but wanted to not get too derailed here

**Review by Twixes — APPROVED** [review]

**Comment by Twixes** [comment]
One small thing is text card headers got a lot taller. So the header overshadows the text visually a bit. Still beta though, so can be improved in a follow-up PR.
big

**Pull request merged**